resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/submit/v2/MountedDependencyManagerImpl.scala

```diff
@@ -0,0 +1,283 @@
+    retrofitClientFactory: RetrofitClientFactory) extends MountedDependencyManager {
+  private val OBJECT_MAPPER = new ObjectMapper().registerModule(new DefaultScalaModule)
+
+  private def localUriStringsToPaths(uris: Seq[String]): Iterable[File] = {
+    KubernetesFileUtils.getOnlySubmitterLocalFiles(uris)
+      .map(Utils.resolveURI)
+      .map(uri => new File(uri.getPath))
+  }
+  private def localJars: Iterable[File] = localUriStringsToPaths(sparkJars)
+  private def localFiles: Iterable[File] = localUriStringsToPaths(sparkFiles)
+
+  override def uploadJars(): String = doUpload(localJars, "uploaded-jars")
+  override def uploadFiles(): String = doUpload(localFiles, "uploaded-files")
+
+  private def doUpload(files: Iterable[File], fileNamePrefix: String): String = {
+    val filesDir = Utils.createTempDir(namePrefix = fileNamePrefix)
+    val filesTgz = new File(filesDir, s"$fileNamePrefix.tgz")
+    Utils.tryWithResource(new FileOutputStream(filesTgz)) { filesOutputStream =>
+      CompressionUtils.writeTarGzipToStream(filesOutputStream, files.map(_.getAbsolutePath))
+    }
+    // TODO provide credentials properly when the staging server monitors the Kubernetes API.
+    val kubernetesCredentialsString = OBJECT_MAPPER.writer()
+      .writeValueAsString(KubernetesCredentials(None, None, None, None))
+    val labelsAsString = OBJECT_MAPPER.writer().writeValueAsString(podLabels)
+
+    val filesRequestBody = RequestBody.create(
+      okhttp3.MediaType.parse(MediaType.MULTIPART_FORM_DATA), filesTgz)
+
+    val kubernetesCredentialsBody = RequestBody.create(
+      okhttp3.MediaType.parse(MediaType.APPLICATION_JSON), kubernetesCredentialsString)
+
+    val namespaceRequestBody = RequestBody.create(
+      okhttp3.MediaType.parse(MediaType.TEXT_PLAIN), podNamespace)
+
+    val labelsRequestBody = RequestBody.create(
+      okhttp3.MediaType.parse(MediaType.APPLICATION_JSON), labelsAsString)
+
+    val service = retrofitClientFactory.createRetrofitClient(
+      stagingServerUri,
+      classOf[ResourceStagingServiceRetrofit],
+      stagingServiceSslOptions)
+    val uploadResponse = service.uploadResources(
+      labelsRequestBody, namespaceRequestBody, filesRequestBody, kubernetesCredentialsBody)
+    val rawSecret = getTypedResponseResult(uploadResponse)
+    rawSecret
+  }
+
+  override def configurePodToMountLocalDependencies(
+      driverContainerName: String,
+      initContainerSecret: Secret,
+      initContainerConfigMap: ConfigMap,
+      originalPodSpec: PodBuilder): PodBuilder = {
+    val sharedVolumeMounts = Seq[VolumeMount](
+      new VolumeMountBuilder()
+        .withName(DOWNLOAD_JARS_VOLUME_NAME)
+        .withMountPath(jarsDownloadPath)
+        .build(),
+      new VolumeMountBuilder()
+        .withName(DOWNLOAD_FILES_VOLUME_NAME)
+        .withMountPath(filesDownloadPath)
+        .build())
+
+    val initContainers = Seq(new ContainerBuilder()
+      .withName("driver-init")
```

Comment:
`spark-driver-init` ?  thinking to if somehow there are other init containers on the pod